File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/dialogue/model/ConversationSessionId.java
package com.study.webflux.rag.domain.dialogue.model;

import java.util.UUID;

public record ConversationSessionId(
	String value
) {
	public ConversationSessionId {
		if (value == null || value.isBlank()) {
			throw new IllegalArgumentException("sessionId cannot be null or blank");
		}
		if (value.length() > 128) {
			throw new IllegalArgumentException("sessionId too long");
		}
	}

	public static ConversationSessionId of(String value) {
		return new ConversationSessionId(value);
	}

	public static ConversationSessionId generate() {
		return new ConversationSessionId(UUID.randomUUID().toString());
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/dialogue/model/ConversationSessionId.kt
package com.miyou.app.domain.dialogue.model

import java.util.UUID

data class ConversationSessionId(
    val value: String,
) {
    init {
        require(value.isNotBlank()) { "sessionId cannot be null or blank" }
        require(value.length <= 128) { "sessionId too long" }
    }

    companion object {
        @JvmStatic
        fun of(value: String): ConversationSessionId = ConversationSessionId(value)

        @JvmStatic
        fun generate(): ConversationSessionId = ConversationSessionId(UUID.randomUUID().toString())
    }

    fun value(): String = value
}
